libraries/node-core-library/src/LockFile.ts

```diff
@@ -85,6 +86,42 @@ export class LockFile {
     throw new Error(`File locking not implemented for platform: "${process.platform}"`);
   }
 
+  /**
+   * Attempts to create the lockfile.
+   * Will continue to loop at every 100ms until the lock becomes available or the maxWaitMs is surpassed.
+   * @remarks This function is subject to starvation, whereby it does not ensure that the process that has been
+   *          waiting the longest to acquire the lock will get it first. This means that a process could theoretically
+   *          wait for the lock forever, while other processes skipped it in line and acquired the lock first.
+   */
+  public static acquire(resourceDir: string, resourceName: string, maxWaitMs?: number): Promise<LockFile> {
+    const interval: number = 100;
+    const startTime: number = Date.now();
+
+    const retryLoop: () => Promise<LockFile> = () => {
```

Comment:
>const retryLoop: () => Promise<LockFile> = () => { [](start = 4, length = 50)

(off topic)

Hey @iclanton,

What do you think of this notation versus a nested function?  The compiler code uses nested functions like crazy, but we pretty much never use them for some reason.  (Maybe a good reason?)